### PR TITLE
Adding time difference route

### PR DESCRIPTION
**Pull request by KGmajor**
Things to do:
Add a new route that displays the time differences between two timezones. 

How to test:
http://0.0.0.0:5000/time-difference?here=America/Chicago&there=America/Denver

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from flask import Flask, request
-import datetime
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
```
Comment (courtneycodes95) [inline app.py:3]: Remember to update the README to include dateutil under Applicable Documentation.

File: app.py [hunk 2]
```diff
@@ -50,7 +51,21 @@ def _get_local_time():
             return f"Could not process time information. {e}", 404
 
 
-#TODO Add a route that displays the current time for all US timezones.
+@app.route('/time-difference', methods=['GET', 'POST','PUT'])
+def time_diff():
```
Comment (courtneycodes95) [inline app.py:55]: It might be helpful to add your note about how to test the route in a comment here for others reading the code directly.

File: app.py [hunk 3]
```diff
@@ -50,7 +51,21 @@ def _get_local_time():
...
+    try:
```
Comment (courtneycodes95) [inline app.py:56]: Great use of error handling!

File: app.py [hunk 4]
```diff
@@ -50,7 +51,21 @@ def _get_local_time():
...
+        if request.args:
+            h = request.args.get('here')
+            t = request.args.get('there')
+
+        u = pytz.timezone('utc').localize(datetime.utcnow())
```
Comment (courtneycodes95) [inline app.py:61]: Lines 61 to 63 are a little hard to follow. Can we use different variable names to describe the transformations that are happening?

**Review by courtneycodes730 — CHANGES_REQUESTED** [review]
Great job overall! The route accomplishes exactly what it needs to and the error messages are very helpful. I left a few suggestions for you to take a look at.

File: app.py (hunk 3 above)
Comment (KGmajor) [inline app.py:56]: The pattern of this api seems to have docstrings, I suggest adding one to make it easier to tell what this does.

File: app.py [hunk 5]
```diff
@@ -50,7 +51,21 @@ def _get_local_time():
             return f"Could not process time information. {e}", 404
 
 
-#TODO Add a route that displays the current time for all US timezones.
+@app.route('/time-difference', methods=['GET', 'POST','PUT'])
```
Comment (KGmajor) [inline app.py:54]: I'm confused about the multiple request methods, should they all be accepted here?

File: app.py [hunk 6]
```diff
@@ -50,7 +51,21 @@ def _get_local_time():
...
+def time_diff():
+    try:
+        if request.args:
+            h = request.args.get('here')
```
Comment (KGmajor) [inline app.py:58]: I think that 'here' and 'there' are a little confusing, I suggest maybe 'timezone1' and 'timezone2'

File: app.py (hunk 4 above)
Comment (KGmajor) [inline app.py:61]: I think it might be more clear if 'u' was a bit more descriptive of the result.
